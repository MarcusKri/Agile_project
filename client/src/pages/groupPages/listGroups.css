/* PS !IMPORTANT IS USED TO OVERRIDE BOOTSTRAP WHICH HAS !IMPORANTANT AS DEFAULT ON MANY ELEMENTS */

/* MAIN PARENT THAT DISPLAYS FLEX */
.listGroupParent {
	width: 100%;
	display: flex;
}

/* MINE GRUPPER  */
.listCardParent {
	width: 100%;
	display: flex;
	justify-content: center;
	flex-wrap: wrap;
}
.listCard {
	width: 100% !important; 
}
.listGroupsParent{
	display: flex;
	width: 45%;
	justify-content: center;
}
.cardImg {
	width: 100%;
	height: 200px;
	overflow: hidden;
}
.cardImg img {
	width: 100% !important;
}
.listCardBody {
	text-align: left;
}
/* BTN */
.listBtn {
	background-color: #9699d5 !important;
	border: 0px !important;
	width: 100%;
}
.listBtnParent {
	width: 100%;
}

#gruppeID {
	width: 100%;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
}

/*  */
@media only screen and (max-width: 1028px) {
	.listCard {
		width: 80% !important;
	}
	.listGroupsParent{
		display: flex;
		width: 100%;
		padding: 10px 10%;
		justify-content: center;
	}
}

