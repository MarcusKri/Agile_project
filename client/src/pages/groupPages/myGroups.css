/* GIVE FLEXBOX */
.flex{
	display: flex;
}
/* MISSION CARD */
#missionCard{
	min-height: 450px;
	padding: 0px !important;
}
/* PARENT TO GROUP PAGE */
.groupPageContainer {
	width: 100%;
	display: flex;
	background-color: #f1f1f1;
	height: 100%;
}
.groupPageContainer::-webkit-scrollbar {
    display: none;
}
/* GROUPS DISPLAY */
.mainDisplayParent {
	width: 100%;
	display: flex;
	flex-direction: column;
	align-items: center;
	overflow-y: scroll;
	overflow-x: hidden;
	background-color: #f1f1f1;
}
.mainDisplayParent::-webkit-scrollbar {
    display: none;
}
.mainDisplay {
	margin: 15px;
	display: flex;
	width: 100%;
	position: relative;
}
.chatDisplay {
	width: 100%;
	justify-content: center;
	align-items: center;
}
.card {
	flex-direction: row;
	padding: 5px;
	margin: 10px;
}
#greenBackgorund {
	background-color: #232b2b;
}
#h1title {
	color: white;
}
label {
	padding: 12px 12px 12px 0;
	display: inline-block;
}
input[type="text"],
select,
textarea,
text {
	width: 100%;
	padding: 12px;
	border: 1px solid #ccc;
	border-radius: 4px;
	resize: vertical;
}
.groupName {
	width: 100%;
	padding: 12px;
	border: 1px solid #ccc;
	border-radius: 4px;
	resize: vertical;
}
