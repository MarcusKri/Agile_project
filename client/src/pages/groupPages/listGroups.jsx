import "bootstrap/dist/css/bootstrap.min.css";
import React from "react";
import { Button, Card } from "react-bootstrap";
import "./listGroups.css";

export function GroupCard({ group: { groupName, course, campus, workForm, anonymity, imageFile } }) {
	function alerMemberShip() {
		// Gives an replacement alert, further functions would be required to acturally add to group. Which we didnt have time/resources to do.
		alert("Du er n√• medlem"); 
	}

	return (
		//LIST GROUPS LAYOUT
		<div className="listGroupsParent">
			<Card className="listCard">
				<div className="cardImg">
					<Card.Img
						variant="bottom"
						src={imageFile}
						alt="Card image"
						style={{ width: "10rem" }}
						className="col"
					/>
				</div>
				<Card.Body className="listCardBody">
					<Card.Text class="d-flex flex-column" id={"cardGroups"}>
						<h3>
							<b>Navn:</b> {groupName}{" "}
						</h3>{" "}
						<b>Campus:</b> {campus} <b>Studie:</b> {course} <b>Arbeidsform:</b>
						{workForm} <b>Anonymitet:</b> {anonymity}
					</Card.Text>
				</Card.Body>
				<div class="listBtnParent">
					<Button type="button" variant="success" className="listBtn" onClick={alerMemberShip}>
						Bli medlem
					</Button>
				</div>
			</Card>
		</div>
	);
}
