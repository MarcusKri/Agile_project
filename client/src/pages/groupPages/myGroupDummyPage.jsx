import "bootstrap/dist/css/bootstrap.min.css";
import React from "react";
import { Button, Card, Nav } from "react-bootstrap";
import myGroups from "../../assets/images/headline-images/myGroups.png";
import ChatBubble from "../chatBubble/chatBubble";
import "../myPage/css/tabs.css";
import { HeadLine } from "../shared/headLine.jsx";
import "./listGroups.css";
import "./myGroups.css";
import AsideNav from "./navigation/asideNav.jsx";
import { NavBar } from "/src/components/navbar/navBar";
import codePicture from "/src/pages/myPage/cards/tabsContent/img/code.jpg";
import codePicture2 from "/src/pages/myPage/cards/tabsContent/img/smidig.jpeg";
import codePicture3 from "/src/pages/myPage/cards/tabsContent/img/peopleworking.jpg";
import { useLoader } from "/src/pages/shared/useLoader";
import { fetchJSON } from "/src/helpers/fetchJSON";
import ErrorHandler from "/src/pages/shared/errorHandler";

export function GroupPage({ myPoints }) {
	const { loading, error } = useLoader(async () => {
		return await fetchJSON("/api/login");
	});

	if (loading) {
		return <></>;
	}

	if (error) {
		return <ErrorHandler />;
	}
	return (
		<div className="mainSection">
			<NavBar myPoints={myPoints} grupper="Grupper" />
			<div className="flex">
				<AsideNav />
				<div className={"groupPageContainer"}>
					<div className={"mainDisplayParent"}>
						<HeadLine headLine="MINE GRUPPER" headImage={myGroups} />
						<div className={"mainDisplay listGroupParent"}>
							<DummyGroupPage />
						</div>
					</div>
				</div>
			</div>
			<ChatBubble />
		</div>
	);
}

//CREATES A DUMMY GROUP PAGE WITH THE GROUPS THE USER IS CONNECTED TO

function DummyGroupPage() {
	return (
		<div className="tabsParent">
			<div className="content-tabs">
				<div className="groupsTabs">
					<div className="listGroupsParent">
						<Card className="listCard">
							<div className="cardImg">
								<Card.Img
									variant="bottom"
									src={codePicture}
									alt="Card image"
									style={{ width: "10rem" }}
									className="col"
								/>
							</div>
							<Card.Body className="listCardBody">
								<Card.Text class="d-flex flex-column" id={"cardGroups"}>
									<h3>
										<b>Navn:</b> UX meetup
									</h3>{" "}
									<b>Campus:</b> Stavanger <b>Studie:</b> Frontend <b>Arbeidsform:</b>
									Fyisk <b>Anonymitet:</b> Ikke anonym
								</Card.Text>
							</Card.Body>
							<div class="listBtnParent">
								<Nav.Link href="/chat">
									<Button type="button" variant="success" className="listBtn">
										Chat
									</Button>
								</Nav.Link>
							</div>
						</Card>
					</div>
					<div className="listGroupsParent">
						<Card className="listCard">
							<div className="cardImg">
								<Card.Img
									variant="bottom"
									src={codePicture2}
									alt="Card image"
									style={{ width: "10rem" }}
									className="col"
								/>
							</div>
							<Card.Body className="listCardBody">
								<Card.Text class="d-flex flex-column" id={"cardGroups"}>
									<h3>
										<b>Navn:</b> Vi er kodere
									</h3>{" "}
									<b>Campus:</b> Bergen <b>Studie:</b> Ã˜konomi <b>Arbeidsform:</b>
									Digitalt <b>Anonymitet:</b> anonym
								</Card.Text>
							</Card.Body>
							<div className="listBtnParent">
								<Nav.Link href="/chat">
									<Button type="button" variant="success" className="listBtn">
										Chat
									</Button>
								</Nav.Link>
							</div>
						</Card>
					</div>
					<div className="listGroupsParent">
						<Card className="listCard">
							<div className="cardImg">
								<Card.Img
									variant="bottom"
									src={codePicture3}
									alt="Card image"
									style={{ width: "10rem" }}
									className="col"
								/>
							</div>
							<Card.Body className="listCardBody">
								<Card.Text class="d-flex flex-column" id={"cardGroups"}>
									<h3>
										<b>Navn:</b> Smidig Prosjekt
									</h3>{" "}
									<b>Campus:</b> Oslo <b>Studie:</b> Hr og Ledelse <b>Arbeidsform:</b>
									Fyisk <b>Anonymitet:</b> Ikke anonym
								</Card.Text>
							</Card.Body>
							<div className="listBtnParent">
								<Nav.Link href="/chat">
									<Button type="button" variant="success" className="listBtn">
										Chat
									</Button>
								</Nav.Link>
							</div>
						</Card>
					</div>
				</div>
			</div>
		</div>
	);
}
