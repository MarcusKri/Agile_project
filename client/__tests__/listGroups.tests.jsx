import React from "react";
import ReactDOM from "react-dom";
import { act, Simulate } from "react-dom/test-utils";
import { ListGroups } from "../src/pages/groupPages/listGroups";
import { GroupsApiContext } from "../src/pages/groupPages/groupsApiContext";

describe("List Groups component", () => {
	it("shows loading screen", () => {
		const domElement = document.createElement("div");
		ReactDOM.render(<ListGroups />, domElement);
		expect(domElement.innerHTML).toMatchSnapshot();
	});

	it("shows groups", async () => {
		const groups = [{ groupName: "domba" }, { groupName: "kalkutta" }];
		const domElement = document.createElement("div");
		await act(async () => {
			ReactDOM.render(
				<GroupsApiContext.Provider value={{ listGroups: () => groups }}>
					<ListGroups />
				</GroupsApiContext.Provider>,
				domElement,
			);
		});
		expect(Array.from(domElement.querySelectorAll("h3")).map((e) => e.innerHTML)).toEqual([
			"<b>Navn:</b> domba ",
			"<b>Navn:</b> kalkutta ",
		]);
		expect(domElement.innerHTML).toMatchSnapshot();
	});

	it("shows error message", async () => {
		const domElement = document.createElement("div");
		await act(async () => {
			const listGroups = () => {
				throw new Error("Something went wrong");
			};
			ReactDOM.render(
				<GroupsApiContext.Provider value={{ listGroups }}>
					<ListGroups />
				</GroupsApiContext.Provider>,
				domElement,
			);
		});
		expect(domElement.querySelector("#error-text").innerHTML).toEqual("Error: Something went wrong");
		expect(domElement.innerHTML).toMatchSnapshot();
	});
});
