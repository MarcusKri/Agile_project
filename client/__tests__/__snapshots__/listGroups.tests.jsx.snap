// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Groups component shows error message 1`] = `"<div><h1>Erorr....</h1><div id=\\"error-text\\">Error: Something went wrong</div></div>"`;

exports[`List Groups component shows groups 1`] = `"<div class=\\"listCardParent\\"><div class=\\"listGroupsParent\\"><div class=\\"listCard card\\"><div class=\\"cardImg\\"><img class=\\"card-img-bottom col\\" alt=\\"Card image\\" style=\\"width: 10rem;\\"></div><div class=\\"listCardBody card-body\\"><p class=\\"d-flex flex-column\\" id=\\"cardGroups\\"><h3><b>Navn:</b> domba </h3> <b>Campus:</b>  <b>Studie:</b>  <b>Arbeidsform:</b> <b>Anonymitet:</b> </p></div><div class=\\"listBtnParent\\"><button type=\\"button\\" class=\\"listBtn btn btn-success\\">Bli medlem</button></div></div></div><div class=\\"listGroupsParent\\"><div class=\\"listCard card\\"><div class=\\"cardImg\\"><img class=\\"card-img-bottom col\\" alt=\\"Card image\\" style=\\"width: 10rem;\\"></div><div class=\\"listCardBody card-body\\"><p class=\\"d-flex flex-column\\" id=\\"cardGroups\\"><h3><b>Navn:</b> kalkutta </h3> <b>Campus:</b>  <b>Studie:</b>  <b>Arbeidsform:</b> <b>Anonymitet:</b> </p></div><div class=\\"listBtnParent\\"><button type=\\"button\\" class=\\"listBtn btn btn-success\\">Bli medlem</button></div></div></div></div>"`;

exports[`List Groups component shows loading screen 1`] = `"<div>Loading.....</div>"`;
